model_analyzer/config/input/config_command.py

```diff
@@ -95,40 +96,137 @@ def set_config_values(self, args):
+                          yaml_config: Dict[str, List]) -> None:
         for key, value in self._fields.items():
             self._fields[key].set_name(key)
-            if key in args:
-                self._check_for_duplicate_profile_models_option(
-                    yaml_config, key)
-                self._fields[key].set_value(getattr(args, key))
-            elif yaml_config is not None and key in yaml_config:
-                self._fields[key].set_value(yaml_config[key])
+            config_value = self._get_config_value(key, args, yaml_config)
+
+            if config_value:
+                self._fields[key].set_value(config_value)
             elif value.default_value() is not None:
                 self._fields[key].set_value(value.default_value())
             elif value.required():
                 flags = ', '.join(value.flags())
                 raise TritonModelAnalyzerException(
                     f'Config for {value.name()} is not specified. You need to specify it using the YAML config file or using the {flags} flags in CLI.'
                 )
-        self._preprocess_and_verify_arguments()
-        self._autofill_values()
 
-    def _check_for_duplicate_profile_models_option(self,
-                                                   yaml_config: Dict[str, List],
-                                                   key: str) -> None:
-        if yaml_config is not None and key in yaml_config and key == 'profile_models':
+    def _get_config_value(
+            self, key: str, args: Namespace,
+            yaml_config: Optional[Dict[str, List]]) -> Optional[Any]:
+        if key in args:
+            return getattr(args, key)
+        elif yaml_config is not None and key in yaml_config:
+            return yaml_config[key]
+        else:
+            return None
+
+    def _check_for_duplicate_profile_models_option(
+            self, args: Namespace, yaml_config: Optional[Dict[str,
+                                                              List]]) -> None:
+        key_in_args = 'profile_models' in args
+        key_in_yaml = yaml_config is not None and 'profile_models' in yaml_config
+
+        if key_in_args and key_in_yaml:
             raise TritonModelAnalyzerException(
                 f'\n The profile model option is specified on both '
                 'the CLI (--profile-models) and in the YAML config file.'
                 '\n Please remove the option from one of the locations and try again'
             )
 
+    def _check_for_multi_model_incompatability(
+            self, args: Namespace, yaml_config: Optional[Dict[str,
+                                                              List]]) -> None:
+        if not self._get_config_value(
+                'run_config_profile_models_concurrently_enable', args,
+                yaml_config):
+            return
+
+        self._check_search_mode(args, yaml_config)
+
+    def _check_search_mode(self, args: Namespace,
```

Comment:
this function needs a better more specific name. This is not a generic check -- it is very specific (including the print!)

Reply:
Updated name